src/services/formatting/smartIndenter.ts

```diff
@@ -10,24 +10,24 @@ namespace ts.formatting {
 
         export function getIndentation(position: number, sourceFile: SourceFile, options: EditorOptions): number {
             if (position > sourceFile.text.length) {
-                return 0; // past EOF
+                return getBaseIndentation(options); // past EOF
             }
 
             // no indentation when the indent style is set to none,
             // so we can return fast
             if (options.IndentStyle === IndentStyle.None) {
-                return 0;
+                return getBaseIndentation(options);
             }
 
             const precedingToken = findPrecedingToken(position, sourceFile);
             if (!precedingToken) {
-                return 0;
+                return getBaseIndentation(options);
             }
 
             // no indentation in string \regex\template literals
             const precedingTokenIsLiteral = isStringOrRegularExpressionOrTemplateLiteral(precedingToken.kind);
             if (precedingTokenIsLiteral && precedingToken.getStart(sourceFile) <= position && precedingToken.end > position) {
-                return 0;
+                return getBaseIndentation(options);

```

Comment:
Per the comment 3 lines above this, you shouldn't indent inside a string literal or regular expression - even in a script block. This will mess up the contents and change its meaning.